source/reflect.h2

```diff
@@ -490,9 +535,12 @@ type_declaration: @copyable type =
     add_member: (inout this, source: std::string_view)
     = {
         decl := parse_statement(source);
-        if !decl || !n*.add_type_member(decl) {
-            error( std::string("error attempting to add member:\n") + source );
-        }
+        require( decl as bool, 
+                 "the provided source string is not a valid statement");
+        require( decl*.is_declaration(), 
+                 "cannot add a member that is not a declaration");
```

Comment:
Another issue.
I just merged #887, which includes commit 5541a752444c0ed89f4e6bf884b3d508d9f0764f, into my working branch.
Apparently, that made `parse_statement` return a `nullptr` value.
After the first `require` above, this one dereferences a null `decl`.

Reply:
I have similar issues in my code.
I'm using it as I would a contract without continuation.